src/main/java/com/google/research/bleth/simulator/Board.java

```diff
@@ -0,0 +1,87 @@
+package com.google.research.bleth.simulator;
+
+import com.google.common.base.Preconditions;
+import com.google.common.collect.ArrayTable;
+import com.google.common.collect.ImmutableList;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+/** A container for all the agents, representing their locations, either real or estimated. */
+public class Board {
+    private final int rowNum;
+    private final int colNum;
+    private ArrayTable<Integer, Integer, ArrayList<Agent>> matrix;
+
+    /**
+     * Create an empty board for storing agents' locations.
+     * A board can represent either the real agents' locations in the simulation,
+     * or the estimated locations from the resolver's point of view.
+     * @param rows is number of rows.
+     * @param cols is number of columns.
+     */
+    public Board(int rows, int cols) {
+        rowNum = rows;
+        colNum = cols;
+        matrix = ArrayTable.create(IntStream.range(0, rows).boxed().collect(Collectors.toList()),
+                                   IntStream.range(0, cols).boxed().collect(Collectors.toList()));
+        for (int row = 0; row < rows; row++) {
+            for (int col = 0; col < cols; col++) {
+                matrix.set(row, col, new ArrayList<Agent>());
+            }
+        }
+    }
+
+    /**
+     * Check if a location is within the board boundaries.
+     * @param location is the location to check if valid.
+     * @return true if the location is valid, false otherwise.
+     */
+    public boolean isLocationInvalid(Location location) {
+        Preconditions.checkNotNull(location);
```

Comment:
Consider statically importing Preconditions.checkNotNull() (and any other methods you use from this class: their names are sufficiently clear to not require the class-name context).